plugins/org.locationtech.udig.project.ui/src/org/locationtech/udig/project/ui/viewers/MapViewer.java

```diff
@@ -258,90 +256,81 @@ else if ( site instanceof IEditorSite){
     public void openContextMenu() {
         final Menu contextMenu = getMenu();
         final Widget control = getControl();
-        if( control != null && !control.isDisposed() && contextMenu != null ){
-            control.getDisplay().asyncExec( new Runnable(){
+        if (control != null && !control.isDisposed() && contextMenu != null) {
+            control.getDisplay().asyncExec(new Runnable() {
+                @Override
                 public void run() {
-                    if( control == null || control.isDisposed() ){
+                    if (control == null || control.isDisposed()) {
                         return;
                     }
                     contextMenu.setVisible(true);
-                }                
+                }
             });
         }
     }
+
     /**
      * Accept the provided tool; call setContext; and activate the tool.
      * <p>
-     * The tool is responsible for registering any mouse listeners it
-     * requires when setActive( true ) is called. When the tool
-     * is replaced setActive( false ) will be called allowing you to
+     * The tool is responsible for registering any mouse listeners it requires when setActive( true
+     * ) is called. When the tool is replaced setActive( false ) will be called allowing you to
      * clean up your listeners.
      * </p>
+     *
      * @param tool
      */
-//    public void setModalTool( ModalTool tool ) {
-//        IToolManager tools = ApplicationGIS.getToolManager();
-//        
-//        if (activeTool != null) {
-//            // ask the current tool to stop listening etc...
-//            activeTool.setActive(false);
-//            activeTool = null;
-//        }
-//        if(tools.getActiveTool() != null ){
-//        	ModalTool globalTool = (ModalTool) tools.getActiveTool();
-//        	globalTool.setActive(false);
-//        }
-//        
-//        if( tool == null ){
-//            return;
-//        }
-//        activeTool = tool;
-//        activeTool.setContext(getToolContext());
-//        activeTool.setActive(true); // this should register itself with the tool manager
-//        
-//        
-//        // this was normally handled by the ToolProxy which we cannot get a hold of
-//        String currentCursorID = activeTool.getCursorID();
-//		Cursor toolCursor = tools.findToolCursor(currentCursorID);
-//		getToolContext().getViewportPane().setCursor(toolCursor);
-//    }
```

Comment:
I'm not sure about what you mean with reactivation functions. Looks like the whole message has a different behaviour than before. Can you explain why?